### PR TITLE
Gcm2

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Reworked as discussed in https://github.com/OP-TEE/optee_os/pull/1908

### CODE REVIEW

File: core/include/crypto/internal_aes-gcm.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+				      const void *src, size_t len, void *dst,
+				      const void *tag, size_t tag_len);
+
+void internal_aes_gcm_inc_ctr(struct internal_aes_gcm_ctx *ctx);
+
+/*
+ * Internal weak functions that can be overridden with hardware specific
+ * implementations.
+ */
+void internal_aes_gcm_next_ctr(struct internal_aes_gcm_ctx *ctx);
```
[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply core/include/crypto/internal_aes-gcm.h]: Indeed, I'll fix.

**Comment by jenswi-linaro** [comment]
Rebased and tags applied

**Comment by jforissier** [comment]
Can you fix the checkpatch warning please?

**Comment by jenswi-linaro** [comment]
Reformatted "core: import GHASH acceleration routines" commit message

**Comment by jforissier** [comment]
Thanks!

**Pull request merged**